tools/FhirSchemaManager/Commands/ApplyCommand.cs

```diff
@@ -4,16 +4,104 @@
-            Console.WriteLine($"--fhir-server {fhirServer}");
-            Console.WriteLine($"--version {version}");
+            schemaClient = new SchemaClient(fhirServer);
+
+            var region = new Region(
+                          0,
+                          0,
+                          Console.WindowWidth,
+                          Console.WindowHeight,
+                          true);
+
+            try
+            {
+                List<AvailableVersion> availableVersions = await schemaClient.GetAvailability();
+
+                if (availableVersions?.Any() != true || availableVersions.Count == 1)
+                {
+                    CommandUtils.PrintError(Resources.AvailableVersionsDefaultErrorMessage);
+                    return;
+                }
+                else
+                {
+                    // Removing the current version
+                    availableVersions.RemoveAt(0);
+                }
+
+                availableVersions = availableVersions.Where(availableVersion => availableVersion.Id <= version)
+                    .ToList();
+
+                foreach (AvailableVersion availableVersion in availableVersions)
+                {
+                    string script = await schemaClient.GetScript(availableVersion.Script);
+
+                    await ValidateVersion(availableVersion.Id);
+
+                    // check if the record for given version exists in failed status
+                    SchemaDataStore.ExecuteQuery(connectionString, string.Join(SchemaDataStore.DeleteQuery, availableVersion.Id), availableVersion.Id);
+
+                    // Execute script
+                    SchemaDataStore.ExecuteQuery(connectionString, script, availableVersion.Id);
+
+                    // Update version status to complete state
+                    SchemaDataStore.ExecuteUpsertQuery(connectionString, availableVersion.Id, "complete");
+
+                    Console.WriteLine(string.Format(Resources.SuccessMessage, availableVersion.Id));
+                }
+            }
+            catch (SchemaManagerException ex)
+            {
+                CommandUtils.PrintError(ex.Message);
+                return;
+            }
+            catch (HttpRequestException)
+            {
+                CommandUtils.PrintError(string.Join(Resources.RequestFailedMessage, fhirServer));
+                return;
+            }
+            catch (SqlException ex)
+            {
+                CommandUtils.PrintError(string.Format(Resources.QueryExecutionExceptionMessage, ex.Message));
+                return;
+            }
+        }
+
+        private static async Task ValidateVersion(int version)
+        {
+            CompatibleVersion compatibleVersion = await schemaClient.GetCompatibility();
+
+            // check if version lies in the compatibility range
+            if (!Enumerable.Range(compatibleVersion.Min, compatibleVersion.Max).Contains(version))
```

Comment:
Range will generate the items, its probably more efficient here to just use a logic check